lib/grape-active_model_serializers/formatter.rb

```diff
@@ -33,15 +33,19 @@ def fetch_serializer(resource, options)
           # use serializer specified by options
           serializer = options[:serializer]
 
-          if serializer.nil?
+          begin
             # fetch serializer leverage AMS lookup
             serializer = ActiveModel::Serializer.serializer_for(resource)
             # if grape version exists, attempt to apply version namespacing
             serializer = namespace_serializer(serializer, options[:version])
-          end
+          rescue LoadError
+            # attempt to fetch serializer through namespace inference
+            serializer = namespace_inferred_serializer(options)
+          end if serializer.nil?
```

Comment:
I would extract this into a method, especially that inside a method you can `rescue` without a `begin` since you can do that on the entire method.